src/MLJCatBoostInterface.jl

```diff
@@ -54,11 +59,10 @@ function prepare_input(X)
     table_input = Tables.columntable(X)
     columns = Tables.columnnames(table_input)
 
-    order_factor_ix = get_dtype_feature_ix(table_input, OrderedFactor)
-    new_columns = Dict([col => MMI.int(table_input[col]) for col in
-                                                             columns[order_factor_ix]]...)
+    order_factor_cols = columns[get_dtype_feature_ix(table_input, OrderedFactor)]
+    new_columns = Dict([col => MMI.int(table_input[col]) for col in order_factor_cols]...)
```

Comment:
I'm a bit nervous about use of dictionary here, because the order is unpredictable. This order conceivably matters: In trees, when determining which feature to split on there can be ties (draws). Maybe in that case the order of the features matters??

Is there a good reason for not using a named tuple here?

Reply:
We already depend on OrderedCollections for OrderedDict's btw, so we could use that

Reply:
There was not a good reason. I have updated it to a named tuple.